python/plugins/processing/algs/qgis/RandomPointsOnLines.py

```diff
@@ -0,0 +1,257 @@
+                                                         self.tr('Minimum distance between points'),
+                                                         0, self.INPUT, False, 0, 1000000000))
+        self.addParameter(QgsProcessingParameterNumber(self.MAXTRIESPERPOINT,
+                                                       self.tr('Maximum number of attempts per point'),
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       10, False, 1, 1000))
+        self.addParameter(QgsProcessingParameterNumber(self.RANDOM_SEED,
+                                                       self.tr('Integer seed (for random)'),
+                                                       QgsProcessingParameterNumber.Integer,
+                                                       None, True, 1, 1000000000))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Random points'),
+                                                            type=QgsProcessing.TypeVectorPoint))
+        self.addOutput(QgsProcessingOutputNumber(self.OUTPUT_POINTS,
+                                                 self.tr('Number of point generated')
+                                                 ))
+
+    def processAlgorithm(self, parameters, context, feedback):
+        source = self.parameterAsSource(parameters, self.INPUT, context)
+        if source is None:
+            raise QgsProcessingException(self.invalidSourceError(parameters,
+                                                                 self.INPUT))
+
+        pointCount = self.parameterAsInt(parameters, self.POINTS_NUMBER,
+                                         context)
+        minDistance = self.parameterAsDouble(parameters, self.MIN_DISTANCE,
+                                             context)
+        randSeed = self.parameterAsInt(parameters, self.RANDOM_SEED, context)
+        MaxTriesPerPoint = self.parameterAsInt(parameters,
+                                               self.MAXTRIESPERPOINT,
+                                               context)
+        fields = QgsFields()
+        fields.append(QgsField('id', QVariant.Int, '', 10, 0))
+
+        (sink, dest_id) = self.parameterAsSink(parameters, self.OUTPUT,
+                                               context, fields,
+                                               QgsWkbTypes.Point,
+                                               source.sourceCrs(),
+                                               QgsFeatureSink.RegeneratePrimaryKey)
+        if sink is None:
+            raise QgsProcessingException(
+                self.invalidSinkError(parameters, self.OUTPUT))
+
+        totNPoints = 0  # The total number of points generated
+        missedPoints = 0  # The number of misses (too close)
```

Comment:
Would be nice to make this an algorithm output too - maybe even two outputs (one for total number of misses, one for number of lines with misses)

Reply:
Will fix - probably tomorrow...

Reply:
Added as outputs (POINTS_MISSED and LINES_WITH_MISSED_POINTS).
I also added an output for the number of null or empty geometries (LINES_WITH_EMPTY_OR_NO_GEOMETRY) and